src/extension.ts

```diff
@@ -63,14 +63,19 @@ export function activate(context: vscode.ExtensionContext): void {
         context.subscriptions.push(tree);
         context.subscriptions.push(vscode.window.registerTreeDataProvider('azureFunctionsExplorer', tree));
 
-        const templateData: TemplateData = new TemplateData(context.globalState);
-
         const actionHandler: AzureActionHandler = new AzureActionHandler(context, outputChannel, reporter);
+
+        const funcPortalTemplateDataTask: Promise<TemplateData | undefined> = tryGetTemplateDataFromFuncPortal(actionHandler, context.globalState);
+        const cachedTemplateDataTask: Promise<TemplateData | undefined> = tryGetTemplateDataFromCache(actionHandler, context.globalState);
```

Comment:
Calling all 3 functions regardless of failure trivializes the telemetry on the tryGetTemplateDataFrom**** events.  I think it could be useful to see how often the portal/cache breaks and when we have to fall back on the stored templates.

I guess it could also be a somewhat expensive operation since you're digging through an array of all the different templates.

Reply:
Each event reports if it succeeds/fails - so we still get the information you want.

However, I agree that it's not the most efficient. I'm going to keep it as-is for 0.6.1 since I've already done quite a bit of testing and don't want to make changes and re-test before releasing, but I'll fix when I merge this to master